src/averageAges.js

```diff
@@ -55,10 +77,36 @@
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const mothers = people.filter(mother => people.some(person =>
+    person.mother === mother.name));
+
+  let kids = people.filter(mummy => mummy.mother != null);
+
+  if (onlyWithSon === true) {
+    kids = people.filter(son => son.mother != null && son.sex === 'm');
```

Comment:
The 'onlyWithSon' parameter is used correctly to filter children. However, the conditions could be simplified for better readability. For example, you can combine the null check and the sex check in a single filter operation.